csa-runner-app/src/main/java/com/farao_community/farao/swe_csa/app/dichotomy/SweCsaRaoValidator.java

```diff
@@ -0,0 +1,93 @@
+import com.powsybl.openrao.data.raoresultjson.RaoResultImporter;
+import com.farao_community.farao.dichotomy.api.NetworkValidator;
+import com.farao_community.farao.dichotomy.api.exceptions.ValidationException;
+import com.farao_community.farao.dichotomy.api.results.DichotomyStepResult;
+import com.farao_community.farao.rao_runner.api.resource.RaoRequest;
+import com.farao_community.farao.rao_runner.api.resource.RaoResponse;
+import com.farao_community.farao.rao_runner.starter.RaoRunnerClient;
+import com.farao_community.farao.swe_csa.api.results.CounterTradeRangeActionResult;
+import com.farao_community.farao.swe_csa.api.results.CounterTradingResult;
+import com.farao_community.farao.swe_csa.app.dichotomy.variable.MultipleDichotomyVariables;
+import com.farao_community.farao.swe_csa.app.rao_result.RaoResultWithCounterTradeRangeActions;
+import com.powsybl.iidm.network.Network;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+
+public class SweCsaRaoValidator implements NetworkValidator<RaoResponse> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SweCsaRaoValidator.class);
+    private final RaoRunnerClient raoRunnerClient;
+    private String requestId;
+    private String networkFileUrl;
+    private String cracFileUrl;
+    private Crac crac;
+    private String raoParametersUrl;
+    private MultipleDichotomyVariables counterTradingValue;
+    private List<String> frEsCnecs;
+    private List<String> ptEsCnecs;
+
+    public SweCsaRaoValidator(RaoRunnerClient raoRunnerClient, String requestId, String networkFileUrl, String cracFileUrl, Crac crac, String raoParametersUrl, Pair<List<String>, List<String>> cnecs) {
+        this.raoRunnerClient = raoRunnerClient;
+        this.requestId = requestId;
+        this.networkFileUrl = networkFileUrl;
+        this.cracFileUrl = cracFileUrl;
+        this.crac = crac;
+        this.raoParametersUrl = raoParametersUrl;
+        this.frEsCnecs = cnecs.getLeft();
+        this.ptEsCnecs = cnecs.getRight();
+    }
+
+    @Override
+    public DichotomyStepResult<RaoResponse> validateNetwork(Network network, DichotomyStepResult<RaoResponse> dichotomyStepResult) throws ValidationException {
+        RaoRequest raoRequest = new RaoRequest(requestId, networkFileUrl, cracFileUrl, raoParametersUrl);
+        try {
+            LOGGER.info("RAO request sent: {}", raoRequest);
+            RaoResponse raoResponse = raoRunnerClient.runRao(raoRequest);
+            LOGGER.info("RAO response received: {}", raoResponse);
+            RaoResult raoResult = new RaoResultImporter().importRaoResult(new URL(raoResponse.getRaoResultFileUrl()).openStream(), crac);
+            RaoResultWithCounterTradeRangeActions raoResultWithCounterTradeRangeActions = this.createRaoResultWithCtRa(raoResult);
```

Comment:
i think here we need flow RAO + voltage monitoring + angle monitoring
Counter trade range action are needed at the end just before exporting results